src/views/virtualmachines/actions/components/DeleteVMModal/DeleteVMModal.tsx

```diff
@@ -84,9 +84,13 @@ const DeleteVMModal: FC<DeleteVMModalProps> = ({ isOpen, onClose, vm }) => {
     }
 
     if (!location.pathname.endsWith('/search')) {
-      navigate(
-        `/k8s/${lastNamespacePath}/${VirtualMachineModelRef}${location.search}${location.hash}`,
-      );
+      const cluster = getCluster(vm);
```

Comment:
Is this logic going to be used anywhere else? If yes, then we should move it to a hook that returns a function that can be called wherever. It would get some of the cluster-specific logic out of this component.

Reply:
No it wil not be used anywhere. We redirect only here. I could even redirect to the all-clusters vm list and don't even use this cluster variable.